pkg/kubectl/cmd/set/set_env.go

```diff
@@ -309,21 +309,22 @@ func (o *EnvOptions) RunEnv(f cmdutil.Factory) error {
 		FilenameParam(enforceNamespace, &o.FilenameOptions).
 		Flatten()
 
-	if o.Local {
-		b = b.Local(f.ClientForMapping)
-	} else {
+	if !o.Local {
 		b = b.
 			LabelSelectorParam(o.Selector).
 			ResourceTypeOrNameArgs(o.All, o.Resources...).
 			Latest()
+	} else {
+		b = b.Local()
 	}
 
 	o.Infos, err = b.Do().Infos()
 	if err != nil {
 		return err
 	}
 	patches := CalculatePatches(o.Infos, o.Encoder, func(info *resource.Info) ([]byte, error) {
-		_, err := o.UpdatePodSpecForObject(info.VersionedObject, func(spec *v1.PodSpec) error {
+		info.Object = info.AsVersioned()
```

Comment:
this will ignore convert err. when convert error will return `the object is not a pod or does not have a pod template` for users. i think this error cannot help users...

but i am ok use `info.AsVersioned()` in `PrintObject`

Reply:
A convert error can only happen if the object is not a valid type, which means we aren't compiled with it, which means we weren't going to be able to support it here anyway.  We already have types that aren't compiled in, so the helper already has to deal with that.

Reply:
I would say that we should be as explicit as we can when something is not in the known type.  We should probably do that up front in the resource.Builder (have a RequireVersioned() flag) that will accumulate errors for anything that isn't versioned.

However, when doing anything from disk we have to be pretty tolerate of things that aren't versioned (i.e. `kubectl set image -f FILE_THAT_HAS_MIXED_OBJECTS`) - that command shouldn't fail for objects it doesn't recognize.  So tolerating untyped objects in set commands is going to be pretty important.